qdax/core/containers/mapelites_repertoire.py

```diff
@@ -386,7 +386,7 @@ def init_default(
 
         # default genotypes is all 0
         default_genotypes = jax.tree_util.tree_map(
-            lambda x: jnp.zeros(shape=(num_centroids,) + x.shape),
+            lambda x: jnp.zeros(shape=(num_centroids,) + x.shape, dtype=x.dtype),
```

Comment:
Could you just give me a recall on why this is necessary. In which use case can it create issues?

Reply:
by default jnp.zeros returns an array of float64
therefore, before this fix any genotype array inserted in the repertoire was automatically casted to foat64 which led to errors when the genotypes need to contain variables that may not be floating number (e.g. integers) which is the case for ME-PBT for instance

Reply:
alright, thanks for the clarification.